source/table_view.tsx

```diff
@@ -213,6 +209,40 @@ export class TableView extends React.Component<Properties, State> implements
     this.headerRowRef.style.cursor = 'auto';
   }
 
+  private tableUpdated(newOperations: Operation[]): void {
+    const start = Math.max(0, this.state.topRow - 1);
+    const end = (() => {
+      if(this.props.model.rowCount === 0) {
+        return 0;
+      } else if(this.state.topRow + this.state.rowsToShow >=
+          this.props.model.rowCount - 1) {
+        return this.props.model.rowCount - 1;
+      } else {
+        return this.state.topRow + this.state.rowsToShow + 1;
+      }
+    })();
+    for(const operation of newOperations) {
+      if(operation instanceof AddRowOperation ||
+        operation instanceof RemoveRowOperation) {
+        if(start <= operation.index && operation.index <= end) {
+          this.forceUpdate();
+        }
+      } else if(operation instanceof UpdateValueOperation) {
+        if(start <= operation.row && operation.row <= end) {
+          this.forceUpdate();
+        }
+
+      } else if(operation instanceof MoveRowOperation) {
+        if(start <= operation.source && operation.source <= end) {
+          this.forceUpdate();
+        } else if(start <= operation.destination &&
+            operation.destination <= end) {
+          this.forceUpdate();
+        }
+      }
+    }
+  }
```

Comment:
This function will end up doing too many ```forceUpdate```, consider the user resorts the table which has potentially 1000s of operations, instead of doing 1000s of forceUpdate you only need to do a single one..